server/images.go

```diff
@@ -795,45 +796,42 @@ func PruneDirectory(path string) error {
 	return nil
 }
 
-func PushModel(ctx context.Context, name string, regOpts *registryOptions, fn func(api.ProgressResponse)) error {
-	mp := ParseModelPath(name)
+func PushModel(ctx context.Context, name model.Name, opts registryOptions, fn func(api.ProgressResponse)) error {
 	fn(api.ProgressResponse{Status: "retrieving manifest"})
 
-	if mp.ProtocolScheme == "http" && !regOpts.Insecure {
-		return errors.New("insecure protocol http")
-	}
-
-	manifest, _, err := GetManifest(mp)
+	m, err := ParseNamedManifest(name)
 	if err != nil {
-		fn(api.ProgressResponse{Status: "couldn't retrieve manifest"})
 		return err
 	}
 
-	var layers []Layer
-	layers = append(layers, manifest.Layers...)
-	if manifest.Config.Digest != "" {
-		layers = append(layers, manifest.Config)
+	scheme := "https"
+	if opts.Insecure {
```

Comment:
did we want to return an error here if someone runs

```
ollama push http://localhost:8080/user/repo
```

In this case it seems like it will still try and push `https://localhost:8080/user/repo`

Reply:
This is kind of a downside of `model.Name`; it discards the scheme since it's not part of the canonical name

Reply:
Can we fix that before merging this PR? The model package should serve `ollama`'s CLI and API experience/contract, not the other way around

Reply:
Correction, pulling from `http` by specifying it in the model name has _never_ worked, with or without `model.Name`. The behaviour has always been to discard the scheme and use HTTPS unless `--insecure` is set.

```
$ ollama pull http://localhost:8080/library/gemma2
pulling manifest
Error: pull model manifest: Get "https://localhost:8080/v2/library/gemma2/manifests/latest": http: server gave HTTP response to HTTPS client
exit status 1
```

Implementing this would be out of scope of this PR since it affects not only push but also pull and anything that touches a model; it requires redoing how models are stored on disk since the scheme is currently not persistented

Reply:
I mentioned this in the other PR. The `scheme` is definitely part of the name. It's implicit like the default host, namespace (i.e. library), and tag (latest). I have a change for that I can push